straxen/analyses/posrec_comparison.py

```diff
@@ -0,0 +1,97 @@
+                             cmt_version=None,
+                             posrec_algos=('mlp', 'gcn', 'cnn')):
+
+    """
+    Returns the corrected position for each position algorithm available,
+        without the need to reprocess event_basics, as the needed
+        information is already stored in event_posrec_many.
+    
+    :param cmt_version: CMT version to use (it can be a list of same
+        length as posrec_algos, if different versions are required for
+        different posrec algorithms, default 'ONLINE')
+    :param posrec_algos: list of position reconstruction algorithms to
+        use (default ['mlp', 'gcn', 'cnn'])
+    """
+    
+    posrec_algos = strax.to_str_tuple(posrec_algos)
+    
+    if cmt_version is None:
+        fdc_config = None
+        try:
+            fdc_config = context.get_single_plugin(run_id, 'event_positions').config['fdc_map']
+            cmt_version = fdc_config[1][1]
+        except IndexError as e:
+            raise ValueError(f'CMT is not set? Your fdc config is {fdc_config}') from e
+    
+    if hasattr(cmt_version, '__len__') and not isinstance(cmt_version, str) and len(cmt_version) != len(posrec_algos):
+        raise TypeError(f"cmt_version is a list but does not match the posrec_algos ({posrec_algos}) length.")
+        
+    cmt_version = (cmt_version, ) * len(posrec_algos) if isinstance(cmt_version, str) else cmt_version
+    
+    dtype = []
+    
+    for algo in posrec_algos:
+
+        dtype += [
+                ((f'Interaction x-position, field-distortion corrected (cm) - {algo.upper()} posrec algorithm', f'x_{algo}'), 
+                 np.float32),
+                ((f'Interaction y-position, field-distortion corrected (cm) - {algo.upper()} posrec algorithm', f'y_{algo}'), 
+                 np.float32),
+                ((f'Interaction z-position, field-distortion corrected (cm) - {algo.upper()} posrec algorithm', f'z_{algo}'), 
+                 np.float32),
+                ((f'Interaction radial position, field-distortion corrected (cm) - {algo.upper()} posrec algorithm', f'r_{algo}'),
+                 np.float32),
+                ((f'Interaction r-position using observed S2 positions directly (cm) - {algo.upper()} posrec algorithm', f'r_naive_{algo}'),
+                 np.float32),
+                ((f'Correction added to r_naive for field distortion (cm) - {algo.upper()} posrec algorithm',
+                  f'r_field_distortion_correction_{algo}'), np.float32),
+                ((f'Interaction angular position (radians) - {algo.upper()} posrec algorithm', f'theta_{algo}'),
+                 np.float32)]
+        
+    dtype += [(('Interaction z-position using mean drift velocity only (cm)', 'z_naive'), np.float32)]
+    result = np.zeros(len(events), dtype=dtype)
+
+    drift_speed = context.get_single_plugin(run_id, 'event_positions').config['electron_drift_velocity']
+    z_obs = - drift_speed * events['drift_time']
```

Comment:
We need to add the gate drift time in order to compute the correct z-positions. Either in this or another PR. See PAX example: https://github.com/XENON1T/pax/blob/c6c813800c03e834265f6f8c33360310248cdaf4/pax/plugins/interaction_processing/BuildInteractions.py#L46